### PR TITLE
Feat/reward properties

### PR DESCRIPTION
**Pull request by msadura**
### <samp>🤖 Generated by Copilot at 5a79f88</samp>

The pull request adds support for rewards pages in the app, which allow users to offer non-monetary incentives for completing tasks. It modifies the `DocumentPage` component, the `RewardProperties` component, and the `RewardsPageWithProviders` component to handle the rewards data and UI. It also fixes some minor issues with the text, the ESLint configuration, and the error handling.

### WHY
Showing proposal details as focalboard properties

Additionally fixed issues that I found:
- creating new reward from rewards page
- fixed reward blocks CRUD (mixing proposal and reward blocks)
- restored adding new custom reward properties from rewards page table
- 500 error from the api when applications were being loaded from reward page

### CODE REVIEW

**Comment by msadura** [comment]
Screenshot 2023-10-04 at 22 21 44

Screenshot 2023-10-04 at 22 18 58
Screenshot 2023-10-04 at 18 34 47

File: components/[pageId]/DocumentPage/DocumentPage.tsx [hunk 1]
```diff
@@ -26,13 +26,15 @@ import { SnapshotVoteDetails } from 'components/common/CharmEditor/components/in
...
 import { useProposalPermissions } from 'components/proposals/hooks/useProposalPermissions';
+import { RewardProperties } from 'components/rewards/components/RewardProperties/RewardProperties';
 import { useBounties } from 'hooks/useBounties';
 import { useBountyPermissions } from 'hooks/useBountyPermissions';
 import { useCharmEditor } from 'hooks/useCharmEditor';
 import { usePageActionDisplay } from 'hooks/usePageActionDisplay';
 import { useVotes } from 'hooks/useVotes';
 import type { PageWithContent } from 'lib/pages/interfaces';
 import type { PageContent } from 'lib/prosemirror/interfaces';
+import Rewards from 'pages/api/rewards';
```
Comment (motechFR) [inline components/[pageId]/DocumentPage/DocumentPage.tsx:37]: Should delete this

File: components/[pageId]/DocumentPage/DocumentPage.tsx [hunk 2]
```diff
@@ -334,7 +337,16 @@ function DocumentPage({ page, refreshPage, savePage, insideModal, readOnly = fal
                         proposalPage={page}
                       />
                     )}
-                    {(draftBounty || page.bountyId) && (
+                    {(draftBounty || page.bountyId) && isRewardsPage && (
```
Comment (motechFR) [inline components/[pageId]/DocumentPage/DocumentPage.tsx:340]: Can you normalise the language to reward, not Bounty?

File: components/[pageId]/DocumentPage/DocumentPage.tsx [hunk 3]
```diff
@@ -334,7 +337,16 @@ function DocumentPage({ page, refreshPage, savePage, insideModal, readOnly = fal
...
+                      <RewardProperties
+                        rewardId={page.bountyId}
+                        pageId={page.id}
+                        pagePath={page.path}
+                        readOnly={readOnly}
+                        refreshRewardPermissions={() => refreshBountyPermissions()}
+                      />
+                    )}
+                    {(draftBounty || page.bountyId) && !isRewardsPage && (
```
Comment (motechFR) [inline components/[pageId]/DocumentPage/DocumentPage.tsx:349]: Can we port draftBounty over to rewards hook? Ideally we should be able to delete the entire useBounties hook(probably only relevant once we are ready to ship)

File: components/common/BoardEditor/components/properties/UserAndRoleSelect.tsx [hunk 4]
```diff
@@ -196,10 +198,20 @@ export function UserAndRoleSelect({
...
+        _filteredOptions = mappedRoles;
+      }
+
+      if (type === 'user') {
+        _filteredOptions = mappedMembers;
+      }
+
+      if (type === 'roleAndUser') {
+        _filteredOptions = [...mappedMembers, ...mappedRoles];
+      }
```
[the remark this answers is not in the document]
Comment (msadura) [reply components/common/BoardEditor/components/properties/UserAndRoleSelect.tsx:211]: I thought I will need it but I did not after all so I can remove

File: components/[pageId]/DocumentPage/DocumentPage.tsx (hunk 3 above)
Comment (msadura) [reply components/[pageId]/DocumentPage/DocumentPage.tsx:349]: actually draftBounty is concept that is not even used (I believe it was  a thing in the past and actually there is no way to create draft bounty right now) so I can just remove this check. We will need to work on "draft reward" from scratch (same concept as proposal)

Comment (msadura) [reply components/[pageId]/DocumentPage/DocumentPage.tsx:349]: ah ok I realized draft bounty is used in db cards, I will fix this

**Pull request merged**